tests/test_manageiq_user.py

```diff
@@ -106,55 +109,54 @@ def fail(msg):
     yield miq
 
 
-def create_user_if_not_exist(miq, miq_api_class, the_group):
+def test_create_user_if_not_exist(miq, miq_api_class, the_group):
     miq_api_class.return_value.collections.groups = [the_group]
+    miq_api_class.return_value.collections.users = []
     miq_api_class.return_value.get.return_value = GET_RETURN_VALUES['user_not_exist']
-    miq_api_class.return_value.get.return_value = POST_RETURN_VALUES['created_user']
+    miq_api_class.return_value.post.return_value = POST_RETURN_VALUES['created_user']
 
     result = miq.create_or_update_user(USERID, USERNAME, PASSWORD, GROUP, EMAIL)
     assert result == {
         'changed': True,
-        'msg': "Successfully created the user dkorn: {}".format(POST_RETURN_VALUES['created_user'])
+        'msg': "Successfully created the user testuser: {}".format(POST_RETURN_VALUES['created_user']['results'])
     }
     miq.client.post.assert_called_once_with(
         '{hostname}/api/users'.format(hostname=MANAGEIQ_HOSTNAME),
         action='create',
         resource={'userid': USERID, 'name': USERNAME, 'password': PASSWORD, 'group': {'id': GROUP_ID}, 'email': EMAIL}
     )
 
-def update_user_email_and_name(miq, miq_api_class, the_user, the_group):
+def test_update_user_email_and_name(miq, miq_api_class, the_user, the_group):
```

Comment:
so previously these didn't run, and all other changes are to actually make it work?